Neos.Fusion/Tests/Unit/Core/Parser/ParserTest.php

```diff
@@ -489,17 +489,18 @@ public function prototypeDeclarationAndInheritance(): array
     /**
      * @test
      */
-    public function testPhpArrayKeys()
+    public function typeConversionOnPhpArrayKeys()
     {
         // as a reminder how php arrays work ^^
-        $asArrayKey = function ($val) {
+        $asArrayKey = static function ($val) {
             return array_key_first([$val => 'something']);
         };
 
         self::assertSame(-123, $asArrayKey(-123), 'negative digits stay');
         self::assertSame(-123, $asArrayKey('-123'), 'negative string digits will be converted to int');
 
-        self::assertSame(123, $asArrayKey(123.456), 'floats will be converted to int');
+        // shut-up operator to suppress PHP 8.1 warning "Deprecated: Implicit conversion from float 123.456 to int loses precision"
```

Comment:
finally! thanks php i never thought you were going to fix that ^^

Reply:
Speaking of which… This test tests PHP itself, I almost wanted to delete it. Like this test I once saw (slightly abridged):

```php
$foo = new Foo();
self::assertInstanceOf(Foo::class, $foo);
```

I left it in place, since it _is_ in fact a nice piece of documentation. 🙈

Reply:
^^ jup i made the above tests, as im testing similar stuff with fusion paths and those results would be confusing without knowing that we are `limited` by php arrays...